chore(api): minor improvement to TransactionBuilder API

bochaco opened the pull request.

davidrusu commented on src/builder.rs:
```diff
@@ -154,9 +154,12 @@ impl TransactionBuilder {
-            self = self.add_input_dbc_bearer(&dbc)?;
+    pub fn add_inputs_dbc_bearer<'a>(
+        mut self,
+        dbcs: impl Iterator<Item = &'a Dbc>,
```
Try using the `Borrow` trait so that this is usable by both borrowed and owned DBC's
```suggestion
        dbcs: impl Iterator<Item = Borrow<Dbc>>,
```

bochaco replied:
i'll play with it, i just gave it a quick try and there are some `Sized` trait related errors....

davidrusu replied:
I think you may have to factor out a type variable:
i.e.
```
dbcs: impl Iterator<Item = D> 
...
where D: Borrow<Dbc>
```

bochaco replied:
yes, that seems to do it, i'll send a commit with that change there as well as in another place i found we also require moving args.

bochaco committed:
chore(api): minor improvement to TransactionBuilder API

- TransactionBuilder::add_inputs_dbc_bearer API was
taking ownership of input DBCs unnecessarily.
